src/components/LibraryDetailPage.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from "react";
+import { LibraryData } from "../interfaces";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import { Store } from "redux";
+import { State } from "../reducers/index";
+import LibraryDetailItem from "./LibraryDetailItem";
+import LibraryStageItem from "./LibraryStageItem";
+import Form from "./reusables/Form";
+
+export interface LibraryDetailPageDispatchProps {
+  editStages: (data: FormData) => Promise<void>;
+  fetchLibrary: (uuid: string) => LibraryData;
+}
+
+export interface LibraryDetailPageStateProps {
+  fullLibrary?: LibraryData;
+}
+
+export interface LibraryDetailPageOwnProps {
+  library: LibraryData;
+  store: Store<State>;
+  uuid: string;
+  updateColor: (library_stage: string, registry_stage: string) => void;
+}
+
+export interface LibraryDetailPageState {
+  libraryStage: string;
+  registryStage: string;
+}
+
+export interface LibraryDetailPageProps extends LibraryDetailPageStateProps, LibraryDetailPageDispatchProps, LibraryDetailPageOwnProps {}
+
+export class LibraryDetailPage extends React.Component<LibraryDetailPageProps, LibraryDetailPageState> {
+
+  constructor(props: LibraryDetailPageProps) {
+    super(props);
+    this.submit = this.submit.bind(this);
+    this.renderInfo = this.renderInfo.bind(this);
+    this.renderStages = this.renderStages.bind(this);
+    this.state = { libraryStage: this.props.library.library_stage, registryStage: this.props.library.registry_stage };
+  }
+
+  renderInfo() {
+    let library = this.props.fullLibrary || this.props.library;
```

Comment:
This is a bit weird that you're expecting data from the server and if it's not there then you're expecting data that was passed down to it. But that data was originally from the server at a higher level. What's the difference and why did you put this in?

Reply:
Ok, I was really tired when I wrote this yesterday, so it's entirely possible that as soon as I look at it this morning I'll realize that there was a much better way to do it, but as it currently stands:

1) When `LibrariesList` loads, it A) gets a list of LibraryData from the server (i.e. data about every library), and then B) maps through the list and creates the `LibrariesListItems` by passing one LibraryData to each one.  That's `this.props.library`.

2) So when you open a `LibraryDetailPage` (by expanding a panel), the data you see is coming from `this.props.library`.  (Originally, I was just giving the `LibrariesListItems` minimal data, and then having `LibraryDetailPage` look up the rest of the data when you open the panel.  But that was giving us the weird flash.  And unnecessarily complicating the data flow.)

3) When you edit a library from inside `LibraryDetailPage`, assuming there are no errors, `LibraryDetailPage` fetches the newly-updated information about that one library.  This is `this.props.fullLibrary`.  It uses that information to update the panel color and the label colors, as well as the list of the library's properties (i.e., if you edit the library_stage and/or registry_stage, their values should be updated).  

TLDR: when you first open the panel, `LibraryDetailPage` is getting its data from `this.props.library` (passed down from `LibrariesList`).  It doesn't have anything called `this.props.fullLibrary` yet, since it only gets that after you hit the save button.  If you hit save, `LibraryDetailPage` fetches the new data for this library (with the changes you just made) and uses that to re-render.    

I feel like I am probably missing something obvious.  But the only other ways I see to do this would be A) to have the entire `LibrariesList` re-render every time you edit one library (which--assuming that there will eventually be a LOT of libraries--seems like a recipe for performance issues), or B) to connect `LibrariesListItem` to the store and have it re-render when you edit the library...but that's the same pattern as what I currently have, just one level up, so that doesn't seem better, particularly given that `LibraryDetailPage` already has to be connected to the store anyway.

Reply:
Hmm interesting but I would say a look for another PR.